Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM rust:1.48 as builder
+
+RUN USER=root cargo new --bin lychee
```

Comment:
I don't see the code being copied into the container... Are we just building hello world for now?

Reply:
In this step yes. We just copy the `Cargo.toml` and trigger a build so that we compile only our dependencies.
A bit further down we `ADD . ./` to copy the source code and run the build again. This should only compile lychee itself. A little trick to reduce build times. 
Of course it only helps for local builds because the Github action starts with an empty environment but we could add [caching](https://github.com/marketplace/actions/build-docker-images-using-cache) for that step in the future.

Reply:
Wow, OK. I see. Hacky!

Let's add a comment explaining these shenanigans.

Reply:
True. 😆 It's kind of a standard approach for other package managers, too. Most notably node.js. Then again it's a good idea to comment that nonetheless, which I now did. 👍